src/sdk.ts

```diff
@@ -2,135 +2,219 @@ import { RelayingServices } from './index';
-                chainId: await this.web3Instance.eth.getChainId(),
-                relayVerifierAddress:
-                    this.contracts.addresses.smartWalletRelayVerifier,
-                deployVerifierAddress:
-                    this.contracts.addresses.smartWalletDeployVerifier,
-                smartWalletFactoryAddress:
-                    this.contracts.addresses.smartWalletFactory
-            }
-        );
-        const resolvedConfig = await resolveConfiguration(
-            this.web3Instance.currentProvider as Web3Provider,
-            partialConfig
-        );
-        resolvedConfig.relayHubAddress = this.contracts.addresses.relayHub;
-        return resolvedConfig;
+        try {
+            const partialConfig: Partial<EnvelopingConfig> = mergeConfiguration(
+                envelopingConfig,
+                {
+                    onlyPreferredRelays: true,
+                    preferredRelays: ['http://localhost:8090'],
+                    gasPriceFactorPercent: 0,
+                    relayLookupWindowBlocks: 1e5,
+                    chainId: await this.web3Instance.eth.getChainId(),
+                    relayVerifierAddress:
+                        this.contracts.addresses.smartWalletRelayVerifier,
+                    deployVerifierAddress:
+                        this.contracts.addresses.smartWalletDeployVerifier,
+                    smartWalletFactoryAddress:
+                        this.contracts.addresses.smartWalletFactory
+                }
+            );
+            const { relayHubAddress, ...newPartialConfig } = partialConfig;
+            const resolvedConfig: EnvelopingConfig = await resolveConfiguration(
+                this.web3Instance.currentProvider as Web3Provider,
+                newPartialConfig
+            );
+            resolvedConfig.relayHubAddress =
+                relayHubAddress ?? this.contracts.addresses.relayHub;
+            return resolvedConfig;
+        } catch (error) {
+            console.log(error);
+        }
     }
 
     async initialize(
-        envelopingConfig: Partial<EnvelopingConfig>
+        envelopingConfig: Partial<EnvelopingConfig>,
+        contractAddresses?: RelayingServicesAddresses
     ): Promise<void> {
-        this.developmentAccounts = await this.web3Instance.eth.getAccounts();
-        const provider = new RelayProvider(
-            this.web3Instance.currentProvider as HttpProvider,
-            await this.configure(envelopingConfig)
-        );
-        if (this.account) {
-            provider.addAccount({
-                address: this.account.address,
-                privateKey: Buffer.from(
-                    this.account.privateKey.replaceAll('0x', ''),
-                    'hex'
-                )
-            });
+        try {
+            this.contracts = new Contracts(
+                this.web3Instance,
+                await this.web3Instance.eth.getChainId(),
+                contractAddresses
```

Comment:
what happens if `contractAddresses` is not proivided? (since it is an optional parameter)

original thread [here](https://github.com/infuy/relaying-services-sdk/pull/2#discussion_r700873085)

Reply:
This is fixed, in the constructor of class Contracts, if contractAddresses is not passed the SDK take all contract from the server config.
``` typescript
contractAddresses = contractAddresses ?? <RelayingServicesAddresses>{};
        const contracts: RelayingServicesAddresses =
            getContractAddresses(chainId);
```